lib/agoneum/web/controllers/registration_controller.ex

```diff
@@ -6,7 +6,27 @@ defmodule Agoneum.Web.RegistrationController do
 
   action_fallback Agoneum.Web.FallbackController
 
+  def new(conn, _params) do
+    changeset = Account.change_user(%User{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
   def create(conn, %{"user" => user_params}) do
+    type = conn |> get_format() |> String.to_atom()
+    create(conn, user_params, type)
+  end
+  defp create(conn, user_params, :html) do
+    case Account.create_user(user_params) do
+      {:ok, user} ->
+        conn
+        |> Guardian.Plug.sign_in(user)
+        |> redirect(to: page_path(conn, :index))
+      {:error, changeset} ->
+        conn
+        |> render(:new, changeset: changeset)
+    end
+  end
+  defp create(conn, user_params, :json) do
```

Comment:
I am curious why we went the `encode_and_sign` route and wondering how that differs from [this](http://blog.overstuffedgorilla.com/simple-guardian-api-authentication/) I think I need to read-up a bit 😆